tests/test_protocol.py

```diff
@@ -1441,3 +1441,62 @@ class TestServer(ServerSlice):
     response = await client.test_method()
     assert response.code == 200
     assert response.result == zip_content
+
+
+@pytest.fixture
+async def options_server():
+    @protocol.typedmethod(path="/test", operation="OPTIONS", client_types=["api"])
+    def test_method() -> ReturnValue[bytes]:  # NOQA
+        pass
+
+    class TestServer(ServerSlice):
+        @protocol.handle(test_method)
+        async def test_methodY(self) -> ReturnValue[str]:  # NOQA
+            return ReturnValue(response="content")
+
+    return TestServer(name="testserver")
+
+
+@pytest.fixture
+def options_request(unused_tcp_port):
+    return HTTPRequest(
+        url=f"http://localhost:{unused_tcp_port}/api/v1/test",
+        method="OPTIONS",
+        connect_timeout=1.0,
+        request_timeout=1.0,
+        decompress_response=True,
+    )
+
+
+@pytest.mark.asyncio
+async def test_auth_enabled_options_method(
```

Comment:
It seems that the entire test is the same expect for the auth true/false. I would prefer to use a parameterized function is that is possible.